Tools/Scripts/collect-pgo-profiles

```diff
@@ -0,0 +1,92 @@
+import os
+import shutil
+import shlex
+import subprocess
+import sys
+import tempfile
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s: %(message)s', datefmt='%Y-%m-%d %H:%M:%S')
+logger = logging.getLogger(__name__)
+
+SCRIPT_DIR = os.path.dirname(__file__)
+DEFAULT_RUN_BENCHMARK = os.path.join(SCRIPT_DIR, 'run-benchmark')
+PGO_PROFILE = os.path.join(SCRIPT_DIR, 'pgo-profile')
+
+
+def ensure_empty_directory(path):
+    if os.path.exists(path):
+        if os.path.isdir(path):
+            shutil.rmtree(path)
+        else:
+            os.remove(path)
+    os.makedirs(path, exist_ok=False)
+
+
+def collect_pgo_profiles(harness, benchmark, output_directory, run_benchmark_args, verbose):
+    benchmark_profile_directory = os.path.join(output_directory, benchmark)
+    ensure_empty_directory(benchmark_profile_directory)
+    with tempfile.TemporaryDirectory() as temp_dir:
+        output_file = os.path.join(temp_dir, f'{benchmark}.json')
+        command = [sys.executable, harness, '--plan', benchmark, '--diagnose-directory', benchmark_profile_directory,
+                   '--generate-pgo-profiles', '--http-server-type', 'builtin', '--count', '1',
+                   '--output-file', output_file, *run_benchmark_args]
+        if verbose:
+            command.append('--debug')
+        logger.info(f'Running {benchmark} with {shlex.join(command)}')
+        subprocess.run(command, check=True)
+
+    merge_command = [sys.executable, PGO_PROFILE, 'merge', benchmark_profile_directory]
+    if verbose:
+        merge_command.append('--verbose')
+    subprocess.run(merge_command, check=True)
+
+    return benchmark_profile_directory
+
+
+def main():
+    parser = argparse.ArgumentParser(prog='collect-pgo-profiles',
+                                     allow_abbrev=False,
+                                     description='PGO profile collection script, '
+                                                 'unknown arguments will be passed to run-benchmark harness.')
+    parser.add_argument('-r', '--run-benchmark-harness', required=False, default=DEFAULT_RUN_BENCHMARK,
+                        help=f'Location for run-benchmark harness, defaults to {DEFAULT_RUN_BENCHMARK}')
+    parser.add_argument('-b', '--benchmarks', nargs='+', required=True,
+                        help='Benchmarks to run for PGO profile collection.')
+    parser.add_argument('-o', '--output-directory', required=True,
+                        help='PGO profile output directory.')
+    parser.add_argument('-p', '--compressed-profile-sub-path', required=True,
+                        help='Sub path under output directory for compressed PGO profiles.')
+    parser.add_argument('-v', '--verbose', action='store_true', default=False,
+                        help='Turn on debug logging')
+    args, run_benchmark_args = parser.parse_known_args()
+
+    if args.verbose:
+        logger.setLevel(logging.DEBUG)
+
+    ensure_empty_directory(args.output_directory)
```

Comment:
I do wonder if we should just fail if the output directory is non-empty, rather than implicitly deleting it? But I haven't too much of a stake in this.

Reply:
Adding timestamp would require a fuzzing matching if we want to get the actual output directory, which may have multiple entries with the same prefix.
I guess I can fail if target directory exists but not empty.